overlord/ifacestate/handlers.go

```diff
@@ -428,9 +413,18 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	conns[connRef.ID()] = connState{Interface: plug.Interface}
+	conns[connRef.ID()] = connState{
```

Comment:
If we are storing static attributes this may misbehave when the snap is changed on disk (snap try will be affected by this). How do you plan to handle that?

Reply:
Thanks for pointing that out. I need to consider this, seems like a valid concern.

Reply:
So this is my last concern for this PR.

I think the question goes back to should snapd cache part of snap info from disk and how it should behave when things can change under us (especially in snap try situations)